homeassistant/components/accuweather/sensor.py

```diff
@@ -292,7 +292,7 @@ class AccuWeatherSensorDescription(
         key="Precipitation",
         device_class=SensorDeviceClass.PRECIPITATION,
         name="Precipitation",
-        state_class=SensorStateClass.MEASUREMENT,
+        state_class=SensorStateClass.TOTAL_INCREASING,
```

Comment:
Is this "past xx h precipitation"? Or "precipitation since xx:xx"?
If it is the first, then I think it should be `SensorStateClass.TOTAL` not increasing.

Reply:
It turns out that Accuweather has marked this value as deprecated, but comparing the values, it comes out to be precipitation in the last hour. I will change `state_class` to `total`. Thanks.

![obraz](https://user-images.githubusercontent.com/478555/212956787-948cd18e-d9d3-4f47-88fe-617bbfce1237.png)

Reply:
Just setting `state_class` to `TOTAL` instead of `TOTAL_INCREASING` will also not work.

To have the precipitation accumulate, you need to set `state_class` to `TOTAL` and also set a `last_reset` to the hour the precipitation is for, as explained in the 4th example here: https://developers.home-assistant.io/docs/core/entity/sensor#how-to-choose-state_class-and-last_reset

Reply:
But in this case it is a "sliding" total, so should MEASUREMENT not then be still allowed?

Reply:
I think the state class isn't wrong. I think this needs a different device class. Although it represents an accumulated amount of precipitation, its window is an hour -> mm/h -> Which can be considered a precipitation intensity device class (which would be a measurement).

Reply:
I see, so
```python
        device_class=SensorDeviceClass.PRECIPITATION_INTENSITY,
        state_class=SensorStateClass.MEASUREMENT,
        metric_unit=UnitOfVolumetricFlux.MILLIMETERS_PER_HOUR,
        us_customary_unit=UnitOfVolumetricFlux.INCHES_PER_HOUR,
```

Reply:
I changed the device class to `PRECIPITATION_INTENSITY` and unit but now I get this warning:
```
The unit of sensor.dom_precipitation (mm/h) can not be converted to the unit of previously compiled statistics (mm). Generation of long term statistics will be suppressed unless the unit changes back to mm or a compatible unit.
```
Maybe the best solution would be to replace this entity with a new "last hour precipitation". It would be a breaking change, but migration from `mm` to `mm/h` would also be a breaking change. What do you think?

Reply:
That error is expected, you can resolve the issue by going to developer tools -> stats tab.

Reply:
I think the PR description will also need to be adjusted to "breaking change" with a small note regarding `developer tools -> stats tab`